public/js/navigator-js/integration/ViewRecipe.js

```diff
@@ -38,6 +42,15 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 		toView: function(viewClass) {
 			this._viewClass = viewClass;
 
+			// This is where the recipe adds a mixin depending on the type of viewClass
+			// added.  This allows for different handling for different 'types' of
+			// view recipes.
+
+			// TODO: Find a more secure way to determine whether class
+			// extends react component
+
```

Comment:
Here I ended up splitting into two methods to simplify things.  So it's `toView` and `toComponent` now.  Was going to go down the peer dependency route but testing is an issue.